iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/mapreduce/IcebergInputFormat.java

```diff
@@ -275,10 +277,134 @@ private static void checkResiduals(CombinedScanTask task) {
 
   @Override
   public RecordReader<Void, T> createRecordReader(InputSplit split, TaskAttemptContext context) {
-    return new IcebergRecordReader<>();
+    if (split instanceof IcebergMergeSplit) {
+      return new IcebergMergeRecordReader<>();
+    } else {
+      return new IcebergRecordReader<>();
+    }
   }
 
-  private static final class IcebergRecordReader<T> extends RecordReader<Void, T> {
+  public abstract static class AbstractIcebergRecordReader<T> extends RecordReader<Void, T> {
```

Comment:
should we move this into a standalone file

Reply:
Done.